lib/connection/connection_config.js

```diff
@@ -260,17 +221,29 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo) {
         protocol: proxyProtocol,
         noProxy: noProxy
       };
-    } else {
-      proxy =
-      {
-        host: proxyHost,
-        port: proxyPort,
-        protocol: proxyProtocol,
-        noProxy: noProxy
-      };
+    } 
+  
+    if (!proxy && GlobalConfig.isEnvProxyActive()) {
+      let proxyEnv = Util.getEnvVar('http_proxy') || Util.getEnvVar('https_proxy');
+
+      if (proxyEnv) {
+        proxyEnv = new URL(proxyEnv);
```

Comment:
this will throw `TypeError` if the user specifies the proxy as `hostname:port` instead of `scheme://hostname:port`

Reply:
I will fix this.